src/oop/object_independence/medium/q4/MainTest.java

```diff
@@ -0,0 +1,225 @@
+        assertTrue(testClass.hasReturnType(firstMethod, null, Void.TYPE), String.format(incorrectReturnType, firstMethod, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void portableMusicPlayerClassHasCorrectPlayMethod(String[] value) throws Throwable {
+        Object[][] arguments = {
+                {value, String[].class}
+        };
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(firstMethod, null), String.format(incorrectMethodDefinition, firstMethod, className));
+        assertTrue(testClass.hasModifier(firstMethod, null, "public"), String.format(incorrectModifierMessage, firstMethod, className));
+        assertTrue(testClass.hasReturnType(firstMethod, null, Void.TYPE), String.format(incorrectReturnType, firstMethod, className));
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        int current = (int) testClass.getFieldValue(classInstance, secondFieldName);
+        testClass.callMethod(firstMethod, methodModifiers, classInstance, new Clause[]{
+                        new StringLiteral("Now Playing : " + value[current])
+                },
+                String.join(" ", "Your", firstMethod, "method does not print the correct output."));
+        _assertEquals(current, testClass.getFieldValue(classInstance, secondFieldName),
+                String.join(" ", "Your", firstMethod, "method should not alter the", secondFieldName, "attribute"));
+    }
+
+    @Test
+    public void playNextIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(secondMethod, null), String.format(incorrectMethodDefinition, secondMethod, className));
+        assertTrue(testClass.hasModifier(secondMethod, null, "public"), String.format(incorrectModifierMessage, secondMethod, className));
+        assertTrue(testClass.hasReturnType(secondMethod, null, Void.TYPE), String.format(incorrectReturnType, secondMethod, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void portableMusicPlayerClassHasCorrectPlayNextMethod(String[] value) throws Throwable {
```

Comment:
Same thing, this never appears to test anything other than the index being in position zero.